core/trino-main/src/main/java/io/trino/server/security/KerberosAuthenticator.java

```diff
@@ -204,7 +203,7 @@ T get()
 
     private static <T> T doAs(Subject subject, GssSupplier<T> action)
     {
-        return Subject.doAs(subject, (PrivilegedAction<T>) () -> {
+        return Subject.callAs(subject, () -> {
```

Comment:
Did you skip migrating another `doAs` method intentionally? 
![Screenshot 2024-01-11 at 11 43 30](https://github.com/trinodb/trino/assets/6237050/600e7085-a485-401e-8427-f1274b05a314)

Reply:
Yes. I wanted to apply this for server only